### DISCUSSION TITLE
HP ScanJet Enterprise Flow 5000 s4 not detected on Linux

### DISCUSSION THREAD

**Discussion by Mebus** [cyanfish/naps2#257] [Support]
Hallo,

the HP ScanJet Enterprise Flow 5000 s4 works fine wiht NAPS2 on Windows. I installed the scanner on Fedora Linux 38 using hplip and it works with simple-scan. Unfortunately it doesn't work with NAPS2 7.2.1. I do not see any devices to choose in the scan dialogue. The file ~/.config/naps2/errorlog.txt is empty aswell. What can I do to make the scanner work?

Thanks!

Mebus

**Comment by cyanfish** [comment]
If you're using the flatpak, it doesn't work with 3rd party drivers so you'll need to use the deb/rpm instead. Otherwise you might just need to wait a bit longer for the scanner to show up (there's supposed to be an animated loading icon but it doesn't work properly on every desktop environment).

**Comment by Mebus** [comment]
Hey, 

you're right, there is a gear wheel, that seems to be the "loading" indicator. When the gear disappears, then there is a list of devices. I've installed it as RPM file.

The problem is, when I want to start scanning I get a pop-up saying "SANE error: IoError" and there is nothing in the ~/.config/naps2/errorlog.txt, still.

I found something in the output of `journalctl -u`:

```
Dez 20 17:16:01 knut naps2[34967]: scan/sane/escl.c 428: escl_open() session=(nil)
Dez 20 17:16:01 knut naps2[34967]: scan/sane/escl.c 53: Calling bb_open
Dez 20 17:16:01 knut naps2[34967]: scan/sane/escl.c 74: Calling bb_load EXIT
Dez 20 17:16:01 knut naps2[34967]: scan/sane/escl.c 459: escl_open() calling bb_escl.so PASSED
Dez 20 17:16:01 knut naps2[34967]: scan/sane/escl.c 277: init_options....
Dez 20 17:16:01 knut naps2[34967]: bb_escl.c 1107: bb_open() escl entry.
Dez 20 17:16:01 knut naps2[34967]: bb_escl.c 965: bb_get_scanner_elements.
Dez 20 17:16:01 knut naps2[34967]: http.c 249: http_open() dd=1 channel=HP-ESCL-SCAN handle=0x7ef8380091a0
Dez 20 17:16:01 knut naps2[34967]: io/hpmud/jd.c 548: unable to connect to escl-scan port 80: Connection refused hp:/net/scanjet_flow_5000_s4?ip=127.0.0.1&queue=false
Dez 20 17:16:01 knut naps2[34967]: http.c 263: unable to open HP-ESCL-SCAN channel
Dez 20 17:16:01 knut naps2[34967]: http.c 249: http_open() dd=1 channel=HP-EWS-LEDM handle=0x7ef8380091a0
Dez 20 17:16:01 knut naps2[34967]: io/hpmud/jd.c 534: unable to connect to ledm-scan port 8080: Connection refused hp:/net/scanjet_flow_5000_s4?ip=127.0.0.1&queue=false
Dez 20 17:16:01 knut naps2[34967]: http.c 263: unable to open HP-EWS-LEDM channel
Dez 20 17:16:01 knut naps2[34967]: http.c 249: http_open() dd=1 channel=HP-IPP handle=0x7ef8380091a0
Dez 20 17:16:01 knut naps2[34967]: io/hpmud/jd.c 565: unable to connect to ipp port 80: Connection refused hp:/net/scanjet_flow_5000_s4?ip=127.0.0.1&queue=false
Dez 20 17:16:01 knut naps2[34967]: http.c 263: unable to open HP-IPP channel
Dez 20 17:16:01 knut naps2[34967]: bb_escl.c 977: unable to open http connection hp:/net/scanjet_flow_5000_s4?ip=127.0.0.1&queue=false
Dez 20 17:16:01 knut naps2[34967]: bb_escl.c 1015: bb_get_scanner_elements returning with (stat=1).
Dez 20 17:16:01 knut naps2[34967]: bb_escl.c 1251: bb_open() returning stat=1.
Dez 20 17:16:01 knut naps2[34967]: scan/sane/escl.c 83: Calling escl bb_unload:
```

How can I fix this?

Mebus

**Reply by cyanfish** [reply]
What's the output of `scanimage -L`? Can you try using "ESCL Driver" instead of "SANE Driver" in your NAPS2 profile settings?
